app/src/androidTest/java/com/github/studydistractor/sdp/procrastinationActivity/ProcrastinationActivityActivityTest.kt

```diff
@@ -16,6 +19,25 @@ import org.junit.Test
 import kotlin.contracts.InvocationKind
 import kotlin.contracts.contract
 
+/**
+ * Uses a [ComposeTestRule] created via [createEmptyComposeRule] that allows setup before the activity
+ * is launched via [onBefore]. Assertions on the view can be made in [onAfterLaunched].
+ *
+ * source : https://stackoverflow.com/questions/68267861/add-intent-extras-in-compose-ui-test
+ */
+inline fun <reified A: Activity> ComposeTestRule.launch(
+    onBefore: () -> Unit = {},
+    intentFactory: (Context) -> Intent = { Intent(ApplicationProvider.getApplicationContext(), A::class.java) },
```

Comment:
It better that way. It was a mistake that the launch was in a different file. That way the tests passes in the CI.